web3/_utils/abi.py

```diff
@@ -550,6 +653,13 @@ def __new__(cls, iterable):
 
 
 def abi_sub_tree(data_type, data_value):
+    if (
+        isinstance(data_type, str) and
+        data_type[0] == "(" and
+        isinstance(data_value, tuple)
+    ):
+        return ABITypedData([data_type, data_value])
```

Comment:
Note below how the array-type ABI values have `abi_sub_tree` applied recursively? I was expecting something like that for a tuple. Otherwise, I suspect `data_tree_map` will fail to apply to the types within the tuple (or arrays within the tuple).